Purchases/FoundationExtensions/String+Extensions.swift

```diff
@@ -16,10 +16,41 @@ import Foundation
 
 extension String {
 
+    enum Error: Swift.Error {
+
+        case escapingEmptyString
+        case trimmingEmptyString
+
+    }
+
     func rot13() -> String {
         ROT13.string(self)
     }
 
+    func escapedOrError() throws -> String {
+        let trimmedAndEscaped = self
+            .trimmingCharacters(in: .whitespacesAndNewlines)
+            .addingPercentEncoding(withAllowedCharacters: .urlHostAllowed)!
+
+        guard trimmedAndEscaped.count > 0 else {
+            Logger.warn("Attempting to escape an empty string")
```

Comment:
if we are throwing, should this be `Logger.error`?

Reply:
💯

Reply:
✅